models/demos/patchtst/tt/optimized/streaming.py

```
@@ -0,0 +1,155 @@
+        self.patch_stride = config.patch_stride
+        self.num_channels = config.num_input_channels
+
+        self.buffer = deque(maxlen=self.context_length)
+
+        self.patch_buffer = []
+
+    def add_timestep(self, timestep_data):
+        if timestep_data.shape[0] != self.num_channels:
+            raise ValueError(
+                f"Expected {self.num_channels} channels, got {timestep_data.shape[0]}"
+            )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)